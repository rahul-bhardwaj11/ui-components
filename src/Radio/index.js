import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AntRadio from 'antd/lib/radio';
import 'antd/lib/radio/style/index.css';
import styled from 'styled-components';
import theme from '../styles/theme';
import mixins from '../styles/mixins';

const MtRadioGroup = styled(AntRadio.Group)`
  &.ant-radio-group {
    font-family: inherit;
  }

  .ant-radio-disabled {
    .ant-radio-inner {
      border: 1px solid ${theme.colors.ALTO};
      background-color: ${theme.colors.PEARL};
      &::after {
        background-color: ${props =>
          props.disabled
            ? `${theme.colors.SILVER}`
            : `${theme.colors.PEARL}`} !important;
      }
    }
  }
`;
const MtRadio = styled(AntRadio)`
  .ant-radio-inner {
    width: 14px;
    height: 14px;
    border: 1px solid ${theme.colors.ALTO};
    &::after {
      top: 2px;
      left: 2px;
    }
  }
  .ant-radio-checked {
    .ant-radio-inner {
      border-color: ${theme.colors.INDIGO};
      &::after {
        background-color: ${theme.colors.INDIGO};
      }
    }
  }
  .ant-radio:hover .ant-radio-inner,
  &.ant-radio-wrapper:hover .ant-radio .ant-radio-inner {
    border-color: ${theme.colors.INDIGO};
  }

  .ant-radio-disabled {
    .ant-radio-inner {
      border: 1px solid ${theme.colors.ALTO};
      background-color: ${theme.colors.PEARL};
      &::after {
        background-color: ${props =>
          props.checked && props.disabled
            ? `${theme.colors.SILVER}`
            : `${theme.colors.PEARL}`};
      }
    }
  }
  &.ant-radio-wrapper {
    font-family: inherit;
    &:hover {
      span.ant-radio + * {
        color: ${theme.colors.SHARK};
      }
    }
  }
  &.ant-radio-wrapper-checked {
    span.ant-radio + * {
      ${mixins.blackLink()};
    }
  }
  span.ant-radio + * {
    ${mixins.greyLink()};
  }

  &.ant-radio-wrapper-disabled {
    span.ant-radio + * {
      color: ${theme.colors.OUTER_SPACE};
    }
  }
  &.ant-radio-wrapper.ant-radio-wrapper-disabled:hover span.ant-radio + * {
    color: ${theme.colors.OUTER_SPACE};
  }
`;

class Radio extends Component {
  static propTypes = {
    children: PropTypes.node,
    checked: PropTypes.bool
  };
  render() {
    let { children } = this.props;
    return <MtRadio {...this.props}>{children}</MtRadio>;
  }
}
Radio.Group = MtRadioGroup;
Radio.Button = AntRadio.Button;
export default Radio;
