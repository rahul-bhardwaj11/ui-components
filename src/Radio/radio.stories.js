import React from 'react';

import { storiesOf } from '@storybook/react';
import Radio from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

const stories = storiesOf('Radio', module);
stories.addDecorator(withKnobs);

stories
  .add(
    'Default Radio',
    withInfo('Basic usage of the Radio')(() => (
      <Radio
        checked={boolean('checked', true)}
        disabled={boolean('disabled', true)}
      >
        {text('children', 'Radio')}
      </Radio>
    ))
  )
  .add(
    'Radio Group',
    withInfo('Usage of the RadioGroup')(() => (
      <Radio.Group>
        <Radio value={1}>1</Radio>
        <Radio value={2}>2</Radio>
        <Radio value={3}>3</Radio>
        <Radio value={4}>4</Radio>
      </Radio.Group>
    ))
  );
