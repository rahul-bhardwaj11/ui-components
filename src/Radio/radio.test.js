import React from 'react';
import renderer from 'react-test-renderer';
import Radio from './index';

describe('Radio', () => {
  it('matches default snapshot', () => {
    const tree = renderer.create(<Radio children="Radio" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
