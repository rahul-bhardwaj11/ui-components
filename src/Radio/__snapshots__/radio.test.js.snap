// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio matches default snapshot 1`] = `
<div
  className="sc-bdVaJa kNssj"
>
  <label
    className="ant-radio-wrapper"
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    style={undefined}
  >
    <span
      className="ant-radio"
      style={Object {}}
    >
      <input
        autoFocus={undefined}
        checked={false}
        className="ant-radio-input"
        disabled={undefined}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        readOnly={undefined}
        tabIndex={undefined}
        type="radio"
        value={undefined}
      />
      <span
        className="ant-radio-inner"
      />
    </span>
    <span>
      Radio
    </span>
  </label>
</div>
`;
